client/components/AllPuzzles.js

```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import {connect} from "react-redux"
+import {fetchAllPuzzles} from "../store/puzzles"
+
+
+export class AllPuzzles extends React.Component{
+  constructor(props){
+    super(props)
+  }
+  componentDidMount(){
+    this.props.fetchAllPuzzles()
+
+  }
+  render(){
+    allPuzzles= this.props.puzzle
+    return(
+      <div>
+        {
+          allPuzzles && allPuzzles.map(puzzle=>{
+            <div key={puzzle.id}>
+            <SinglePuzzle puzzle={puzzle}/>
+            </div>
+          })
+
+        }
+      </div>
+    )
+  }
+}
+
+const mapState =(state) => {
+  return {
+    puzzles:state.puzzles.puzzles
```

Comment:
puzzles:state.puzzles.puzzles is technically, fine but kind of confusing.

If you get a chance, would be nice to set it up as --> puzzles: state.puzzles.all (or something)

Just a suggestion for readability, not a necessity. :)

Reply:
Agreed! Renaming to puzzles.allPuzzles for clarity